.circleci/config.yml

```diff
@@ -1,28 +1,71 @@
-version: 2
-jobs:
-  build:
+version: 2.1
+commands:
+  runtests:
+    steps:
+      - checkout
+      - restore_cache: # Download and cache dependencies
+          key: libhoney-java-{{ checksum "pom.xml" }}
+      - run: ./mvnw dependency:go-offline # gets the project dependencies
+      - save_cache: # saves the project dependencies
+          paths:
+            - ~/.m2
+          key: libhoney-java-{{ checksum "pom.xml" }}
+      - run: ./mvnw package # run the actual tests
+executors:
+  java8:
+    working_directory: ~/libhoney-java
+    environment:
+      # Customize the JVM maximum heap limit
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
     docker:
       - image: circleci/openjdk:8-jdk-stretch
-
+  java9:
+    working_directory: ~/libhoney-java
+    environment:
+      # Customize the JVM maximum heap limit
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+    docker:
+      - image: circleci/openjdk:9-jdk-sid
+  java10:
+    working_directory: ~/libhoney-java
+    environment:
+      # Customize the JVM maximum heap limit
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+    docker:
+      - image: circleci/openjdk:10-jdk-sid
+  java11:
     working_directory: ~/libhoney-java
-
     environment:
       # Customize the JVM maximum heap limit
       JVM_OPTS: -Xmx3200m
       TERM: dumb
-
+    docker:
+      - image: circleci/openjdk:11-jdk-stretch
```

Comment:
Is it intentional that 8 and 11 use stretch (oldstable release, not current stable) and 9 and 10 use sid (unstable) in their image choices? If so, could you add a comment saying why? If not, it'd be lovely to omit the version of debian entirely if possible (does `circleci/openjdk:11-jdk` work?) or switch to `buster` (current stable) otherwise.

Reply:
The images are based on what I found available from CircleCI [here](https://circleci.com/docs/2.0/circleci-images/#openjdk). They have recently added some buster images but only for (the newly added) JDK 13/14...I am not sure why they have no images for JDK 12. I did find that list very quirky with regards to the underlying OS. :)

I have gone ahead and changed all the images to be the `circleci/openjdk:<JDK version>-jdk` as requested.